plugins/carbon_black_cloud/icon_carbon_black_cloud/util/utils.py

```diff
@@ -0,0 +1,85 @@
+from typing import List
+from ipaddress import ip_network, ip_address
+from logging import Logger
+import re
+
+
+HOST = "host"
+IP_NETMASK = "ip-netmask"
+IP_RANGE = "ip-range"
+DEVICE_ID = "device-id"
+
+REGEX_HOST = "[a-zA-Z]"
+REGEX_IP_NETMASK = "/"
+REGEX_IP_RANGE = "-"
+
+
+class Util:
+    @staticmethod
+    def determine_agent_type(agent: str) -> str:
+        """
+        determine_agent_type. Determine the type of agent based on its name.
+
+        :param agent: A string representing the name of the agent to classify.
+        :type: str
+
+        :return: The type of agent.
+        :rtype: str
+        """
+
+        if re.search(REGEX_HOST, agent):
+            return HOST
+        if re.search(REGEX_IP_NETMASK, agent):
+            return IP_NETMASK
+        if re.search(REGEX_IP_RANGE, agent):
+            return IP_RANGE
+        try:
+            int(agent)
+            return DEVICE_ID
+        except Exception:  # noqa: B110
+            pass
+        return IP_NETMASK
+
+    @staticmethod
+    def match_whitelist(agent: str, whitelist: List[str], logger: Logger) -> bool:
+        """
+        match_whitelist. Check if the given agent string matches any of the whitelist strings.
+
+        :param agent: The agent string to be checked.
+        :type: str
+
+        :param whitelist: A list of strings to be compared against the agent string.
+        :type: List[str]
+
+        :param logger: A logger object.
+        :type: Logger
+
+        :return: True if the agent string matches any of the whitelist strings, False otherwise.
+        :rtype: bool
+        """
+
+        object_type = Util.determine_agent_type(agent)
+        if object_type in (HOST, DEVICE_ID):
+            if agent in whitelist:
+                logger.info(f" Whitelist matched\n{agent} was found in whitelist")
+                return True
+            else:
+                return False
+
+        # if 1.1.1.1/32 - remove /32
+        trimmed_address = re.sub(r"/32$", "", agent)
+
+        # if contains / we compare explicit matches, but not subnets in subnets
+        if "/" in trimmed_address:
+            return agent in whitelist
```

Comment:
Does this `if` ever get hit? It looks like if `/32` is removed, then `/` is never found in `trimmed_address`.

Are there cases where the ip has `/xx` that != `32`?

Reply:
It's about the subnet mask. It can be for example `/24` 😄